client/src/interface/pages/FileManager/index.js

```diff
@@ -100,15 +96,11 @@ export default class FileManager extends React.Component {
         }, 200);
     };
 
-    handleChangePermission = (id, mode) => {
-        this.changePermission.current.update_id_mode(this.state.cwd, id, mode & 0b111111111, this.session_id)
+    handleChangePermission = (name, mode) => {
+        this.changePermission.current.update_id_mode(this.state.cwd, name, mode, this.session_id)
```

Comment:
pass the full mode over so <ChangePermission> can tell whether the entry is a file or directory. if it is a dir, <ChangePermission> will show the "Apply Recusively" option